trk/src/Screens/GymDaily/index.js

```diff
@@ -1,8 +1,219 @@
+      let climbFrequencyMap = {};
+      let timeFrequencyMap = {};
+
+
+
+      for (const climb of yourClimbs) {
+        const tapsSnapshot = await getTapsBySomeField('climb', climb.id);
+        if (tapsSnapshot && tapsSnapshot.docs) {
+          const taps = tapsSnapshot.docs.map(doc => {
+            const tap = doc.data();
+            tap.timestamp = tap.timestamp.toDate();
+            console.log(`Tap Timestamp: ${tap.timestamp}`);
+            return tap;
+          });
+          totalTaps += taps.length;
+          allTaps = [...allTaps, ...taps];
+
+          latestAscentsCount += taps.filter(tap => tap.timestamp > oneDayAgo).length;
+
+          taps.forEach(tap => {
+            if (!climbFrequencyMap[tap.climb]) {
+              climbFrequencyMap[tap.climb] = { count: 0, name: climb.name, id: climb.id, grade: climb.grade };
+            }
+            climbFrequencyMap[tap.climb].count++;
+          });
+        }
+      }
+
+      for (const tap of allTaps) {
+        if (tap.timestamp) {
+          const hour = tap.timestamp.getHours(); 
+          timeFrequencyMap[hour] = (timeFrequencyMap[hour] || 0) + 1;
+        }
+      }
+
+
+      let peakHour = Object.keys(timeFrequencyMap).reduce((a, b) => timeFrequencyMap[a] > timeFrequencyMap[b] ? a : b, null);
+      setPeakTime(peakHour ? `${peakHour}:00 - ${parseInt(peakHour) + 1}:00` : '');
+      console.log(`peakHour is: ${peakHour}`)
+
+      setAscents(allTaps);
+      setTotalAscents(totalTaps.toString());
+      setLatestAscents(latestAscentsCount.toString());
+
+      let mostCompletedClimb = Object.values(climbFrequencyMap).reduce((a, b) => (a && b && a.count > b.count) ? a : b, null);
+      let leastCompletedClimb = Object.values(climbFrequencyMap).reduce((a, b) => (a && b && a.count < b.count) ? a : b, null);
+
+      if (mostCompletedClimb) {
+        setMostCompleted({ name: mostCompletedClimb.name, id: mostCompletedClimb.id, grade: mostCompletedClimb.grade });
+      } else {
+        setMostCompleted({ name: '', id: '', grade: '' });
+      }
+
+      if (leastCompletedClimb) {
+        setLeastCompleted({ name: leastCompletedClimb.name, id: leastCompletedClimb.id, grade: leastCompletedClimb.grade });
+      } else {
+        setLeastCompleted({ name: '', id: '', grade: '' });
+      }
+
+    };
+
+
+    if (yourClimbs.length > 0) {
+      fetchTapsAndCalculateAscents();
+    }
+  }, [yourClimbs]);
+
+
+  useEffect(() => {
+    const fetchCommentsAndCalculateFeedback = async () => {
+      try {
+        const commentsPromises = yourClimbs.map(climb =>
+          getCommentsBySomeField('climb', climb.id)
+        );
+        const commentsSnapshots = await Promise.all(commentsPromises);
+        let comments = [];
+        commentsSnapshots.forEach(snapshot => {
+          snapshot.docs.forEach(doc => comments.push({ ...doc.data(), timestamp: doc.data().timestamp.toDate() }));
+        });
+
+        setYourComments(comments);
```

Comment:
using the CommentsApi function 'getCommentsBySomeField' we retrieve all the comments that are associated to the climbs you have set. we also get the timestamp of the comment which we need to later defined the most recent (or latest) feedback comment